packages/testing-library/lynx-dom/src/index.ts

```diff
@@ -0,0 +1,393 @@
+  target.console.profileEnd = () => {};
+
+  target.__LoadLepusChunk = __LoadLepusChunk;
+
+  globalThis.onInjectMainThreadGlobals?.(target);
+}
+
+const IGNORE_LIST_GLOBALS = [
+  'globalThis',
+  'global',
+];
+
+export function injectBackgroundThreadGlobals(target?: any) {
+  const {
+    app,
+    performance,
+    CoreContext,
+    JsContext,
+    __LoadLepusChunk,
+  } = createPolyfills();
+  if (typeof target === 'undefined') {
+    target = globalThis;
+  }
+
+  target.timestamp = Date.now();
+  target.__DEV__ = true;
+  target.__PROFILE__ = true;
+  target.__JS__ = true;
+  target.__LEPUS__ = false;
+  target.globDynamicComponentEntry = '__Card__';
+  target.lynxCoreInject = {};
+  target.lynxCoreInject.tt = {
+    _params: {
+      initData: {},
+      updateData: {},
+    },
+  };
+
+  class NodesRef {
+    private readonly _nodeSelectToken: any;
+    private readonly _selectorQuery: any;
+
+    constructor(selectorQuery: any, nodeSelectToken: any) {
+      this._nodeSelectToken = nodeSelectToken;
+      this._selectorQuery = selectorQuery;
+    }
+    invoke() {
+      throw new Error('not implemented');
+    }
+    path() {
+      throw new Error('not implemented');
+    }
+    fields() {
+      throw new Error('not implemented');
+    }
+    setNativeProps() {
+      throw new Error('not implemented');
+    }
+  }
+  const enum IdentifierType {
+    ID_SELECTOR, // css selector
+    REF_ID, // for react ref
+    UNIQUE_ID, // element_id
+  }
+
+  const globalEventEmiter = new EventEmitter();
+  // @ts-ignore
+  globalEventEmiter.trigger = globalEventEmiter.emit;
+  // @ts-ignore
+  globalEventEmiter.toggle = globalEventEmiter.emit;
+  target.lynx = {
+    getNativeApp: () => app,
+    performance,
+    createSelectorQuery: (() => {
+      return {
+        selectUniqueID: function(uniqueId: number) {
+          return new NodesRef({}, {
+            type: IdentifierType.UNIQUE_ID,
+            identifier: uniqueId.toString(),
+          });
+        },
+      };
+    }),
+    getCoreContext: (() => CoreContext),
+    getJSContext: (() => JsContext),
+    getJSModule: (moduleName) => {
+      if (moduleName === 'GlobalEventEmitter') {
+        return globalEventEmiter;
+      } else {
+        throw new Error(`getJSModule(${moduleName}) not implemented`);
+      }
+    },
+  };
+  target.requestAnimationFrame = setTimeout;
+  target.cancelAnimationFrame = clearTimeout;
+
+  target.console.profile = () => {};
+  target.console.profileEnd = () => {};
+
+  // TODO: user-configurable
+  target.SystemInfo = {
+    'platform': 'iOS',
+    'pixelRatio': 3,
+    'pixelWidth': 1170,
+    'pixelHeight': 2532,
+    'osVersion': '17.0.2',
+    'enableKrypton': true,
+    'runtimeType': 'quickjs',
+    'lynxSdkVersion': '3.0',
+  };
+
+  target.__LoadLepusChunk = __LoadLepusChunk;
+
+  globalThis.onInjectBackgroundThreadGlobals?.(target);
+}
+
+export type ThreadType = 'background' | 'main';
+
+export class LynxDOM {
```

Comment:
Let's rename this to LynxToplevel or something. We shouldn't confuse our setup with Web DOM furthermore.

Additionally, I encourage you to add more documentation-comments whenever you declared a new concept. 

I wouldn't even concern about you use AI to generate such doc comments as a starting point, so you can understand how AI read through your code.

Reply:
Good catch, since it is used by vitest/jest to setup their environment:

- https://jestjs.io/docs/configuration#testenvironment-string
- https://vitest.dev/guide/environment

I would like to name it `LynxRuntime` (just like `edge-runtime`) or `LynxEnvorinment` (means can be used as environment), let me use `LynxRuntime` first, please comment it you have any suggestions~